options.go

```diff
@@ -5,13 +5,50 @@ import (
 	"fmt"
 	"net"
 	"os"
+	"strings"
 	"time"
 
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	"github.com/yl2chen/cidranger"
 )
 
+// resolver contains info about a single upstream DNS server.
+type resolver struct {
+	addr  string   //address of the resolver in format ip:port
+	proto []string //list of protocols to use with this resolver, in order of execution
+}
+
+func (r resolver) getAddr() string {
+	return r.addr
+}
+
+func (r resolver) getProto() []string {
+	return r.proto
+}
+
+// tcpOnly forces the resolver to only use tcp, regardless of the user input.
+// Required for backwards compatibility with v1.
+func (r resolver) tcpOnly() {
+	r.proto = []string{"tcp"}
```

Comment:
I'd like treating `tcpOnly` as the default case only. For example: 

input resolvers | with `tcpOnly` | without `tcpOnly`
------------ | ------------- | ------------
1.1.1.1 | **tcp@1.1.1.1:53** | udp+tcp@1.1.1.1:53
udp+tcp@1.1.1.1 | udp+tcp@1.1.1.1:53 | udp+tcp@1.1.1.1:53
udp@1.1.1.1 | udp@1.1.1.1:53 | udp@1.1.1.1:53

Consider the backward compatible case, the input resolvers are plain `ip:port`, the `tcpOnly` flag still takes effect. I believe this strategy has less confusion to the parameter combinations.

Reply:
Done! tcpOnly flag now only applies to the default ip:port case.